include/zenoh-pico/net/private/system.h

```diff
@@ -15,15 +15,19 @@
 #ifndef _ZENOH_NET_PICO_SYSTEM_H
 #define _ZENOH_NET_PICO_SYSTEM_H
 
-#if (ZENOH_LINUX == 1) || (ZENOH_MACOS == 1)
+#if (ZENOH_LINUX == 1) || (ZENOH_MACOS == 1) || (ESP32 == 1) || (ARDUINO_ARCH_ESP32 == 1)
```

Comment:
It would be preferable to use the same naming convention:
ESP32 -> ZENOH_ESP32
ARDUINO_ARCH_ESP32 -> ZENOH_ARDUINO_ESP32

Reply:
Although this might be possible, I think it is better to keep it like this.
   - to use your naming conventions, this needs to be done either in the CMakefile.txt (this is a still pending point) or to add them **manually** in whatever tool for embedded development you use (I am using platformio for example).
   - in my experience, I have already seen several occurrences of people failing to correctly follow the installation instructions, especially if they require manually/additional configurations.

Let me know if you still want me to change them.

Reply:
Based on your reply, I believe we might consider to leave the naming as you proposed. However, it would be great that zenoh-pico could be compiled for ESP32 platform also via the shipped CMakefile, do you think is it feasible? What kind of changes would be required?

Reply:
I was taking a look in the file struct of zenoh-pico, and maybe we can solve this by having a mapping in the [zenoh-pico.h](zenoh-pico.h). Something like:
```
#if (ESP32 == 1) || (ARDUINO_ARCH_ESP32 == 1)
  #define ZENOH_ESP32
#endif
```
This would allow to use your naming conventions in the rest of the code and to avoid issues when compiling with the CMake or when using tools for embedded development like platformio.

Regarding CMake, it is something that I would need to look into, but for sure it is feasible to compile for ESP32 platform via the shipped CMakefile.

Reply:
I like the proposal. I believe the zenoh-pico.h is a good place where to put those kinds of definitions. Please let us know if you need any help with the CMakefile, unfortunately I don't have an ESP32 board at hand to help you with.